rancher/rancher
# [CAPR] Remove `kube-scheduler` and `kube-controller-manager` certificate/key files for regeneration when performing certificate rotation

## Issue: 
https://github.com/rancher/rancher/issues/41125
 
## Problem
The `kube-scheduler` and `kube-controller-manager` are both able to generate self-signed certificates for use as a serving certificate. CAPR uses the certificates generated by the components for healthz probes. After 1 year, the certificates expire, which can lead to CAPR throwing an error that the probes are failing.

## Solution
This PR adds rotation of certificates to the `kube-scheduler` and `kube-controller-manager` certificates by way of the `rotate certificates` functionality.
 
## Testing

## Engineering Testing
### Manual Testing
1. Spin up a `K3s` cluster using CAPR.
1. Check the OpenSSL expiration date of the certificate found at `/var/lib/rancher/k3s/server/tls/kube-scheduler/kube-scheduler.crt`
1. Run the rotate certificates plan
1. Observe that the expiration date of the certificate is "newer" than the old certificate

### Automated Testing
1. While I am unable to (in an automated fashion) test that the certificate file is rotated currently, I added a unit test to the function I modified to ensure it is delivering the correct number of instructions based on node role, and able to handle customization of the `cert-dir` argument.
2. 
## QA Testing Considerations

 
### Regressions Considerations
